client/logic/game/units/unit.gd

```diff
@@ -270,4 +273,4 @@ func _end_move() -> void:
 
 func _set_unit_stars() -> void:
 	if _unit_stats and is_on_terrain():
-		_unit_stats.star_number = get_terrain().properties.defence_level
+		_unit_stats.star_number = _types.terrains[get_terrain().id]["defense"]
```

Comment:
same for the terrain info. Having some class that extends resource for that, will avoid errors.